### PR TITLE
Omit caching valid request with empty content

### LINKED ISSUE #5: BaseSource.scrape() should check size of src_content
Problem: If I get a HTTP 200 with an empty answer I will create an empty file in the cache. Not a big issue but just not very helpful.

When deciding if it should follow up with save cache and extracting data, we should also check for the size of `src_content`. If `src_content` is "empty", do not cache the file or try to extract anything. Let's save those precious CPU cycles!

### PR DESCRIPTION
**Pull request by eddowh**
closes #5

### CODE REVIEW

File: openkongqi/source.py [hunk 1]
```diff
@@ -63,6 +63,11 @@ def scrape(self):
         self._now = datetime.now(self._tz)
         src_content = self.fetch()
         self.save_status()
+        content_size = src_content.headers.get('content-length')
+        src_content_is_empty = content_size == None or content_size == 0
+        if src_content.getcode == 200 and src_content_is_empty:
```
Comment (hrbonz) [inline openkongqi/source.py]: I'd rather see this as a multi-line test.

Comment (eddowh) [inline openkongqi/source.py]: You mean like this:

```python
content_size = src_content.headers.get('content-length')
if src_content.getcode == 200 and (content_size == None or content_size == 0):
    ...
```

Or multiline everything:

```python
if src_content.getcode == 200 and
    (src_content.headers.get('content-length') == 0 or
     src_content.headers.get('content-length') == None):
    ...
```

**Comment by hrbonz** [comment]
I'd personally do the first version because I like to keep method.function calls to a minimum. Both would be correct though.
I would use a variable called `content_length` to keep the HTTP header meaning.

**Comment by eddowh** [comment]
Fixed test + renamed to `content_length`.

**Comment by hrbonz** [comment]
squash the first two commits, revert last commit about PEP8.

**Comment by eddowh** [comment]
Done.

**Comment by eddowh** [comment]
Wow, I squashed my commit where I changed `==` to `is` and pushed it; now your comment is gone...

**Comment by hrbonz** [comment]
Interesting, anyway I'll merge

**Pull request merged**